bin/og-tax-summary.R

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env Rscript
+
+# Read in commandline arguments
+args = commandArgs(trailingOnly=TRUE)
+
+# Get path to the orthogroup gene count summary file
+ogcounts <- args[1]
+# And then the path to the samplesheet
+samples <- args[2]
+
+numsppFilt <- args[3]
+numgrpFilt <- args[4]
+copynumFilt1 <- args[5]
+copynumFilt2 <- args[6]
+
+ogs <- read.delim(ogcounts)
+samples <- read.delim(samples, sep = ",")
+
+colnames(ogs) <- gsub("\\..*", "", colnames(ogs))
+
+# Extract and remove the totals column
+totals <- ogs['Total']
+ogs <- ogs[,-which(colnames(ogs) == 'Total')]
+
+# Convert zero-counts to NA
+ogs[-1][ogs[-1] == 0] <- NA
+
+# Calculate the mean copy number per species
+copynum <- apply(ogs[-1], 1, mean, na.rm=T)
+
+# Convert counts to a binary presence/absence
+ogs[-1][ogs[-1] > 0] <- 1
+
+# For each orthogroup, count the number of species included
+numspp <- rowSums(ogs[-1], na.rm = T)
+
+# Convert species names to taxon group
+for(i in 2:ncol(ogs)){
+    # Identify the species
+    spp <- colnames(ogs)[i]
+    
+    # And the taxonomic group for this species
+    grp <- unique(as.character(samples$taxonomy[which(samples$species == spp)]))
+    
+    # Now replace the species name with group name
+    colnames(ogs)[i] <- grp
+}
+
+# Count number of taxonomic groups included in each orthogroup
+taxcount <- t(rowsum(t(ogs[-1]), 
+              group = colnames(ogs)[-1], 
+              na.rm = T))
+taxcount[taxcount > 1] <- 1
+taxcount <- rowSums(taxcount)
+
+res <- 
+    data.frame(
+        orthogroup = ogs$Orthogroup,
+        num_spp = numspp,
+        total_copy_num = totals,
+        mean_copy_num = copynum,
+        num_tax_grps = taxcount
+    )
+
+# Create the subsets
+extreme_core <- 
```

Comment:
Yeah, I hate the names. Basically I want to use a broad, but conservative set of gene trees for species tree inference (e.g. includes some number of species, some number of taxonomic groups, and some maximum copy number). 

This is in part due to a need to limit the number of trees used in species tree inference (time and computational constraints). So I create a subset for that (the extreme core) and infer MSAs, trim them, and infer gene family trees all as one "tributary" of the workflow. 

But I also want to infer gene family trees (and MSAs, etc) for some of these larger but more time consuming gene families, so I relax (increase) the copy number threshold, and then infer all of these large MSAs and trees separately so that it doesn't hold up the inference of the species tree which we would like to use sooner than later. 

Presumably that second branch of the analysis could be run as an optional step, I just haven't figured out how yet since that doesn't matter for my immediate use case.

Reply:
Okay, I've renamed these to "spptree_core" and "genetree_core" to clearly indicate that one subset is used for species tree inference (as well as gene tree), whereas the other subset is used only for gene tree inference.